components/monday/monday.app.mjs

```diff
@@ -1,11 +1,348 @@
+import mondaySdk from "monday-sdk-js";
+import uniqBy from "lodash.uniqby";
+import map from "lodash.map";
+import flatMap from "lodash.flatmap";
+import constants from "./common/constants.mjs";
+import mutations from "./common/mutations.mjs";
+import queries from "./common/queries.mjs";
+
 export default {
   type: "app",
   app: "monday",
-  propDefinitions: {},
+  propDefinitions: {
+    boardId: {
+      type: "integer",
+      label: "Board ID",
+      description: "The board's unique identifier",
+      async options({ page }) {
+        return this.listBoardsOptions({
+          page: page + 1,
+        });
+      },
+    },
+    boardName: {
+      type: "string",
+      label: "Board Name",
+      description: "The board's name",
+    },
+    boardKind: {
+      type: "string",
+      label: "Board Kind",
+      description: "The board's kind (`public` / `private` / `share`)",
+      options: constants.BOARD_KIND_OPTIONS,
+    },
+    folderId: {
+      type: "integer",
+      label: "Folder ID",
+      description: "Board folder ID",
+      optional: true,
+    },
+    workspaceId: {
+      type: "integer",
+      label: "Workspace ID",
+      description: "Board workspace ID. If you don't specify this field, the board will be created in the **Main Workspace**",
+      optional: true,
+      async options({ page }) {
+        if (page) {
+          return [];
+        }
+        return this.listWorkspacesOptions();
+      },
+    },
+    templateId: {
+      type: "integer",
+      label: "Template ID",
+      description: "Board template ID",
+      optional: true,
+    },
+    groupName: {
+      type: "string",
+      label: "Group Name",
+      description: "The name of the new group",
+    },
+    groupId: {
+      type: "string",
+      label: "Group ID",
+      description: "The group's unique identifier",
+      optional: true,
+      async options({
+        page, boardId,
+      }) {
+        if (page) {
+          return [];
+        }
```

Comment:
Is this `if` statement needed? I tried removing it, and it appears to work the same. It's present in props `workspaceId`, `groupId`, and `itemId`.